metrics-service-ui/data-source-plugin/src/QueryEditor.tsx

```diff
@@ -2,47 +2,68 @@ import defaults from 'lodash/defaults';
 
 import React, { ChangeEvent, PureComponent } from 'react';
 import { LegacyForms } from '@grafana/ui';
-import { QueryEditorProps } from '@grafana/data';
+import { QueryEditorProps, SelectableValue } from '@grafana/data';
 import { DataSource } from './datasource';
 import { defaultQuery, MyDataSourceOptions, MyQuery } from './types';
 
 const { FormField } = LegacyForms;
+const availableClustersEndpoint = 'https://localhost:10010/metrics-service/api/v1/clusters';
 
 type Props = QueryEditorProps<DataSource, MyQuery, MyDataSourceOptions>;
 
 export class QueryEditor extends PureComponent<Props> {
   onQueryTextChange = (event: ChangeEvent<HTMLInputElement>) => {
     const { onChange, query } = this.props;
-    onChange({ ...query, queryText: event.target.value });
+    onChange({ ...query, metricName: event.target.value });
   };
 
-  onConstantChange = (event: ChangeEvent<HTMLInputElement>) => {
-    const { onChange, query, onRunQuery } = this.props;
-    onChange({ ...query, constant: parseFloat(event.target.value) });
-    // executes the query
-    onRunQuery();
+  onClusterChange = (event: SelectableValue<HTMLInputElement>) => {
+    const { onChange, query } = this.props;
+    onChange({ ...query, clusterName: event.target.value });
   };
 
+  async componentDidMount() {
+    fetch(availableClustersEndpoint)
+      .then((resp) => resp.json())
+      .then((clusters) => {
+        let items: JSX.Element[] = [];
```

Comment:
Looks like this can be `const`.
There could be a more clear name than `items` as well, the name could describe the UI element it is holding.

Reply:
Which element would you like to be const?